lib/rubocop/ast/node/ensure_node.rb

```diff
@@ -6,10 +6,31 @@ module AST
     # node when the builder constructs the AST, making its methods available
     # to all `ensure` nodes within RuboCop.
     class EnsureNode < Node
+      DEPRECATION_WARNING_LOCATION_CACHE = [] # rubocop:disable Style/MutableConstant
+      private_const :DEPRECATION_WARNING_LOCATION_CACHE
+
       # Returns the body of the `ensure` clause.
       #
       # @return [Node, nil] The body of the `ensure`.
+      # @deprecated Use `EnsureNode#branch`
       def body
+        first_caller = caller(1..1).first
+
+        unless DEPRECATION_WARNING_LOCATION_CACHE.include?(first_caller)
+          warn '`EnsureNode#body` is deprecated and will be changed in the next major version of ' \
+               'rubocop-ast. Use `EnsureNode#branch` instead to get the body of the `ensure` branch.'
+          warn "Called from:\n#{caller.join("\n")}\n\n"
+
+          DEPRECATION_WARNING_LOCATION_CACHE << first_caller
+        end
+
+        branch
+      end
```

Comment:
I'm not sure if this is overkill?

I thought it'd be useful to output a warning given that the method will no longer work the same post #336 release. At the same time, I am going to update all the @rubocop org repos to use the new method name, so the risk is only to 3rd party extensions that might use `EnsureNode#body`.

I also wanted it to not spam deprecation warnings, hence the cache to make sure each warning is just output once.

Reply:
@Earlopain it looks like `StrictWarnings` is causing the rubocop master specs to fail because of the warning 🤦. Any ideas?

Reply:
Well, this is awkward. I didn't anticipate this interaction.

Currently there is nothing to disable this but it is easy enough to control through environment variable. It already does nothing if `CI` is not set but I don't think that would be appropriate here. I can make a PR tomorrow.

Rails sets `STRICT_WARNINGS=1` in their CI instead of relying on something generic, once again I should just have followed what they do.

Reply:
> I'm not sure if this is overkill?

I don't think so, I like it.